### PR TITLE
Refactors to allow for paged data

### PR DESCRIPTION
**Pull request by madninja**
This adds async and stream support for paged reponses.

It alsorefactors the library to use modules to scope api areas

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -142,98 +64,110 @@ impl Client {
...
+                    (data, client, path),
+                    |(mut data, client, path)| async move {
+                        match data.data.pop() {
+                            Some(entry) => Ok(Some((entry, (data, client, path)))),
+                            None => match data.cursor {
+                                Some(cursor) => {
+                                    let mut data = client
+                                        .fetch_data::<Vec<E>, _>(&path, &[("cursor", &cursor)])
+                                        .await?;
+                                    data.data.reverse();
```
Comment (lthiery) [inline src/lib.rs:117]: What's the reverse for? If you fetch a stream of `blocks` they start from highest and go to lower thanks to the reverse before popping?

**Review by lthiery — APPROVED** [review]
Looks great! I would ask for one more feature from the Oracle Prices module. I need OraclePredictions for the wallet sweep branch I have on `helium-wallet-rs`.


[ I try to replicate the pessimistic implementation of the app by default, which uses the max of predicted and current oracle price to determine how much HNT must be left behind. ]

File: src/validators.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    /// number is unknown at this time.
+    pub active: Option<u64>,
+    pub staked: StakeStats,
+    pub unstaked: StakeStats,
+    pub cooldown: StakeStats,
+}
+
+/// The current staked status for a validator.
+#[derive(Clone, Deserialize, Debug)]
+pub enum StakeStatus {
```
Comment (lthiery) [inline src/validators.rs:37]:
```suggestion
#[serde(rename_all = "lowercase")]
pub enum StakeStatus {
```

File: src/lib.rs (hunk 1 above)
Comment (madninja) [reply src/lib.rs:118]: exactly

**Comment by madninja** [comment]
> Looks great! I would ask for one more feature from the Oracle Prices module. I need OraclePredictions for the wallet sweep branch I have on `helium-wallet-rs`.
> 
> [ I try to replicate the pessimistic implementation of the app by default, which uses the max of predicted and current oracle price to determine how much HNT must be left behind. ]

Added to the latest push here

**Pull request merged**